### ISSUE TITLE
Label icons on main nav bar

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#1857]
In the bottom nav bar on the home screens of the app, we want to show a text label along with the icon for each button.

This should also resolve the lack of such labels for screen readers, which was one of the points noted at https://github.com/zulip/zulip-flutter/issues/535#issuecomment-2973276995 .

### References
* Chat thread with user request: [#mobile > Flutter feedback - Icons in app](https://chat.zulip.org/#narrow/channel/48-mobile/topic/Flutter.20feedback.20-.20Icons.20in.20app/with/2223128)
* Design: [#mobile > Flutter feedback - Icons in app @ 💬](https://chat.zulip.org/#narrow/channel/48-mobile/topic/Flutter.20feedback.20-.20Icons.20in.20app/near/2219685)
